ERCS/erc-draft_pm_capability.md

```diff
@@ -0,0 +1,285 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-03
+requires: 4337, 5792
+---
+
+## Abstract
+
+With [EIP-5792]('./eip-5792.md'), apps can communicate with wallets about advanced features via capabilities. This proposal defines a capability that allows apps to request that [ERC-4337]('./eip-4337.md') wallets communicate with a specified paymaster web service. To support this, we also define a standardized API for paymaster web services.
+
+## Motivation
+
+App developers want to start sponsoring their users' transactions using paymasters. Paymasters are commonly used via web services. However, there is currently no way for apps to tell wallets to communicate with a specific paymaster web service. Similarly, there is no standard for how wallets should communicate with these services. We need both a way for apps to tell wallets to communicate with a specific paymaster web service and a communication standard for wallets to do so.
+
+## Specification
+
+One new [EIP-5792]('eip-5792.md') wallet capability is defined. We also define a standard interface for paymaster web services as a prerequisite.
+
+### Paymaster Web Service Interface
+
+We define two JSON-RPC methods to be implemented by paymaster web services.
+
+#### `pm_getPaymasterStubData`
+
+Returns stub values to be used in paymaster-related fields of an unsigned user operation for gas estimation. Accepts an unsigned user operation, entrypoint address, chain id, and a context object. Paymaster service providers can define fields that app developers should use in the context object. 
+
+##### `pm_getPaymasterStubData` RPC Specification
+
+```typescript
+// [userOp, entryPoint, chainId, context]
+type GetPaymasterStubDataParams = [
+  // Below is specific to Entrypoint v0.6 but this API can be used with other entrypoint versions too
+  {
+    sender: `0x${string}`;
+    nonce: `0x${string}`;
+    initCode: `0x${string}`;
+    callData: `0x${string}`;
+    callGasLimit: `0x${string}`;
+    verificationGasLimit: `0x${string}`;
+    preVerificationGas: `0x${string}`;
+    maxFeePerGas: `0x${string}`;
+    maxPriorityFeePerGas: `0x${string}`;
+    paymasterAndData: `0x${string}`;
+    signature: `0x${string}`;
+  },
+  `0x${string}`,
+  `0x${string}`,
+  Record<string, any>
+];
+
+type GetPaymasterStubDataResult = Record<string, `0x${string}`> // Stub paymaster values
+```
+
+###### `pm_getPaymasterStubData` Example Parameters
+
+```json
+[
+  {
+    "sender": "0x...",
+    "nonce": "0x...",
+    "initCode": "0x",
+    "callData": "0x...",
+    "callGasLimit": "0x...",
+    "verificationGasLimit": "0x...",
+    "preVerificationGas": "0x...",
+    "maxFeePerGas": "0x...",
+    "maxPriorityFeePerGas": "0x...",
+    "paymasterAndData": "0x...",
+    "signature": "0x..."
+  },
+  "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789",
+  "0x2105",
+  {
+    "policyId": "962b252c-a726-4a37-8d86-333ce0a07299"
```

Comment:
what is policyId?

Reply:
this is illustrative. paymaster services should define fields they might look for here.